common/lib/capa/capa/responsetypes.py

```diff
@@ -1967,10 +1993,23 @@ def get_extended_hints(self, student_answers, new_cmap):
         """
         if self.answer_id in student_answers:
             if new_cmap.cmap[self.answer_id]['correctness'] == 'correct':  # if the grader liked the student's answer
-                # Note: using self.id here, not the more typical self.answer_id
-                hints = self.xml.xpath('//numericalresponse[@id=$id]/correcthint', id=self.id)
-                if hints:
-                    hint_node = hints[0]
+                # Answer is not an additional answer.
+                if self.additional_answer_index == -1:
+                    # Note: using self.id here, not the more typical self.answer_id
+                    hints = self.xml.xpath('//numericalresponse[@id=$id]/correcthint', id=self.id)
+                    if hints:
+                        hint_node = hints[0]
+                        new_cmap[self.answer_id]['msg'] += self.make_hint_div(
+                            hint_node,
+                            True,
+                            [student_answers[self.answer_id]],
+                            self.tags[0]
+                        )
```

Comment:
Can we move this `new_cmap[self.answer_id]['msg'] += self.make_hint_div(` outside the if else to reduce the redundancy ?